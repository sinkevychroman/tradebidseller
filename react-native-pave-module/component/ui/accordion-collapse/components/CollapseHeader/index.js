import React from 'react';
import {View} from 'react-native';

const CollapseHeader = React.forwardRef(({children, ...restProps}, ref) => {
  return (
    <View ref={ref} {...restProps}>
      {children}
    </View>
  );
});

export default CollapseHeader;
