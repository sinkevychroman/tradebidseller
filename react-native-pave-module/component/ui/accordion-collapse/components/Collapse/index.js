/**
 * @flow
 */

import React, {useState, useEffect} from 'react';
import {View, TouchableOpacity} from 'react-native';
import CollapseHeader from '../CollapseHeader';
import CollapseBody from '../CollapseBody';

const Collapse = React.forwardRef(
  (
    {
      isCollapsed = false,
      disabled = false,
      onToggle = () => undefined,
      handleLongPress = () => undefined,
      children,
      ...restProps
    },
    ref,
  ) => {
    const [show, setShow] = useState(isCollapsed);
    useEffect(() => {
      setShow(isCollapsed);
    }, [isCollapsed]);
    let header = null;
    let body = null;
    React.Children.forEach(children, child => {
      if (child.type === CollapseHeader) {
        header = child;
      } else if (child.type === CollapseBody) {
        body = child;
      }
    });
    if (header === null) {
      console.warn(
        "header wasn't found to be rendered. Please make sure you have wrapped an CollapseHeader in the Collapse Component.",
      );
      return null;
    } else {
      return (
        <View ref={ref} {...restProps}>
          <TouchableOpacity
            disabled={disabled}
            onPress={() => {
              onToggle(!show);
              setShow(!show);
            }}
            onLongPress={handleLongPress}>
            {header}
          </TouchableOpacity>
          {show && body}
        </View>
      );
    }
    // eslint-disable-next-line prettier/prettier
  }
);

export default Collapse;
