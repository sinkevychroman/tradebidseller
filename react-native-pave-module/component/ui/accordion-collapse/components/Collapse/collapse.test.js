import 'react-native';
import React from 'react';

import Collapse from './index';
import {act, create} from 'react-test-renderer';
import {TouchableOpacity} from 'react-native';
import CollapseHeader from '../CollapseHeader';
import CollapseBody from '../CollapseBody';

describe('Collapse', () => {
  it('Should render Collapse', () => {
    create(
      <Collapse>
        <CollapseHeader>click here</CollapseHeader>
        <CollapseBody>Ta da!</CollapseBody>
      </Collapse>,
    );
  });
  it('Should match snapshot', () => {
    const renderedComponent = create(
      <Collapse>
        <CollapseHeader>click here</CollapseHeader>
        <CollapseBody>Ta da!</CollapseBody>
      </Collapse>,
    );
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });
  it('Should match snapshot while collapsed', () => {
    const renderedComponent = create(
      <Collapse isCollapsed={true}>
        <CollapseHeader>click here</CollapseHeader>
        <CollapseBody>Ta da!</CollapseBody>
      </Collapse>,
    );
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });
  it('Should display body when clicked', () => {
    let renderedComponent;
    act(() => {
      renderedComponent = create(
        <Collapse>
          <CollapseHeader>click here</CollapseHeader>
          <CollapseBody>Ta da!</CollapseBody>
        </Collapse>,
      );
    });
    expect(
      renderedComponent.root.find(node => node.props.children === 'click here'),
    ).toBeDefined();
    const touchable = renderedComponent.root.findByType(TouchableOpacity);
    expect(
      renderedComponent.root.find(node => node.props.children === 'click here'),
    ).toBeDefined();
    expect(() =>
      renderedComponent.root.find(node => node.props.children === 'Ta da!'),
    ).toThrow();
    act(touchable.props.onPress);
    expect(
      renderedComponent.root.find(node => node.props.children === 'Ta da!'),
    ).toBeDefined();
    expect(
      renderedComponent.root.find(node => node.props.children === 'click here'),
    ).toBeDefined();
  });
});
