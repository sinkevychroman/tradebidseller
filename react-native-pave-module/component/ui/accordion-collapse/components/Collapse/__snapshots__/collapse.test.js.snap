// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Should match snapshot 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View>
      click here
    </View>
  </View>
</View>
`;

exports[`Collapse Should match snapshot while collapsed 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View>
      click here
    </View>
  </View>
  <View>
    Ta da!
  </View>
</View>
`;
